blocks/bricks/attention.py

```diff
@@ -605,8 +605,11 @@ def take_glimpses(self, **kwargs):
             kwargs.pop(self.preprocessed_attended_name, None),
             kwargs.pop(self.attended_mask_name, None),
             **dict_union(states, glimpses_needed))
-        if kwargs:
-            raise ValueError("extra args to take_glimpses: {}".format(kwargs))
+        # The following check is too excessive.
+        # there could be context information which is needed by the transition
+        # apply, but is also passed to this method.
+        # if kwargs:
+        #    raise ValueError("extra args to take_glimpses: {}".format(kwargs))
```

Comment:
It is also called from BaseSequenceGenerator.generate which passes to it the arguments with names BaseSequenceGenerator._context_names
This is actuall a call to BaseSequenceGenerator.transition.apply.contexts which is actually a call to
AttentionRecurrent.do_apply_contexts

This returns  `attended` and `attended_mask` which are needed to take a glimpsee but it also includes context supplied by the transition (eg LSTM):
    AttentionRecurrent.transition.apply.contexts
Up to now all basic transition classes supplied in recurrent.py dont have context so the problem 
was left unnotticed. But I have wrote a wrapper for basic transition which adds context (will be published at some point) and this is how I found this bug.